scripts/enrich_match_data.py

```diff
@@ -0,0 +1,242 @@
+import json
+import pickle
+import datetime
+import jsonlines
+import wikipedia
+from wikipedia.exceptions import DisambiguationError, PageError
+from typing import List, Dict
+
+from nltk.corpus import stopwords
+from nltk.tokenize import RegexpTokenizer
+
+from emma.OntoEmma import OntoEmma
+import emma.utils.string_utils as string_utils
+from emma.paths import StandardFilePath
+from emma.kb.kb_utils_refactor import KnowledgeBase
+
+
+# class for enriching entity with synonyms
+class SynonymEnricher:
+    def __init__(self):
+        """
+        Intialize
+        """
+        # Load synonym files
+        paths = StandardFilePath()
+        mesh_syn_file = os.path.join(paths.ontoemma_synonym_dir, 'mesh_synonyms.pickle')
+        dbpedia_syn_file = os.path.join(paths.ontoemma_synonym_dir, 'dbpedia_synonyms.pickle')
+
+        sys.stdout.write("Loading synonym files..\n")
+        self.mesh_synonyms = pickle.load(open(mesh_syn_file, 'rb'))
+        self.dbpedia_synonyms = pickle.load(open(dbpedia_syn_file, 'rb'))
+
+    def add_synonyms_to_entity(self, aliases: List):
+        """
+        Return synonyms of entity
+        :param aliases: entity aliases
+        :return:
+        """
+        # normalize aliases
+        norm_aliases = [string_utils.normalize_string(a) for a in aliases]
+
+        # intialize synonym lists
+        mesh_syns = []
+        dbpedia_syns = []
+
+        # get synonyms from synonym dicts
+        for a in norm_aliases:
+            mesh_syns += self.mesh_synonyms[a]
+            dbpedia_syns += self.dbpedia_synonyms[a]
+
+        return list(set(mesh_syns)), list(set(dbpedia_syns))
+
+
+# class for querying entities in Wikipedia to extract context
+class WikipediaEnricher:
+    def __init__(self):
+        """
+        Intialize queryer
+        :param kb_name: name of KB
+        :param kb_path: path to KB file (any format readable by OntoEmma)
+        :param out_path: output path of enriched KB
+        :param restrict_to_ids: set restricting KB ids to query
+        """
+        # Initialize wikipedia queryer
+        wikipedia.set_lang('en')
+        wikipedia.set_rate_limiting(True, min_wait=datetime.timedelta(0, 0, 2000))
+        self.wiki_dict = dict()
+
+        self.tokenizer = RegexpTokenizer(r'[A-Za-z\d]+')
+        self.STOP = set(stopwords.words('english'))
+
+    @staticmethod
+    def get_summary(query_text: str):
+        """
+        Get summary from hit article
+        :param query_text:
+        :return:
+        """
+        try:
+            return wikipedia.summary(query_text, sentences=1)
+        except DisambiguationError:
+            return ''
+        except PageError:
+            return ''
+        except Exception:
+            print('Unknown exception!')
+            return ''
+
+    def search_term(self, query_text: str):
+        """
+        Search query text in wikipedia
+        :param query_text:
+        :return:
+        """
+        if query_text not in self.wiki_dict:
```

Comment:
It's not clear why we need this here but not in `get_summary`. Should we use a similar dictionary in `get_summary` to avoid querying the same `query_text` multiple times?